### PR TITLE
Deferred and declined

### PR DESCRIPTION
**Pull request by jordan-beard**
This pull request contains the initial implementation for the declined and deferred user journeys.

### CODE REVIEW

**Review by MartinWheelerMT — COMMENTED** [review]
I'll approve this at the moment, but this does not fit the pattern for TryGet extensions, which should return a boolean value with an out parameter.

I would be tempted to not have this and to use IDictionary's TryGetValue extension instead, and consume using a ternary operator if the out value needs to be set to a property value.

File: ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/Shared/_ProjectHeader.cshtml [hunk 1]
```diff
@@ -12,6 +12,7 @@
     </div>
     @if (!string.IsNullOrEmpty(Context.Request.Query["rd"]))
     {
+
```
Comment (MartinWheelerMT) [inline ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/Shared/_ProjectHeader.cshtml:15]: unnecessary whitespace

**Review by MartinWheelerMT — APPROVED** [review]
Generally happy with the changes here.  Has been commented that the missing tests will be completed, a long with a few clean up tweaks.  I'm happy for this to be deployed as is and we can complete those in a further PR

File: ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/Shared/_ProjectHeader.cshtml (hunk 1 above)
Comment (jordan-beard) [reply ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/Shared/_ProjectHeader.cshtml]: Either I've addressed this already or its due to the way github displays things.
